cli/cmd/create.go

```diff
@@ -48,259 +37,109 @@ var createCmd = &cobra.Command{
-		builderConfig.WithProfile(slice.Profile)
-	}
-	ab := artifact_builder.NewArtifactBuilder(isDryRun).WithConfig(builderConfig).WithBackend(backend)
-
-	workspaceRepo := createWorkspaceRepo(isDryRun, backend)
-
-	err = verifyTFEBacklog(ctx, workspaceRepo)
-	if err != nil {
-		return err
-	}
-
-	stackService := stackservice.NewStackService().WithBackend(backend).WithWorkspaceRepo(workspaceRepo)
-
-	existingStacks, err := stackService.GetStacks(ctx)
+	ctx := cmd.Context()
+	err = validate(
+		validateGitTree(happyClient.HappyConfig.GetProjectRoot()),
+		validateTFEBackLog(ctx, dryRun, happyClient.AWSBackend),
+		validateStackNameAvailable(ctx, happyClient.StackService, stackName, force),
+		validateStackExistsCreate(ctx, stackName, dryRun, happyClient),
+		validateECRExists(ctx, stackName, dryRun, terraformECRTargetPathTemplate, happyClient),
+		validateImageExists(ctx, createTag, skipCheckTag, happyClient.ArtifactBuilder),
+	)
 	if err != nil {
-		return err
-	}
-	existingStack, stackAlreadyExists := existingStacks[stackName]
-	if stackAlreadyExists && !force {
-		return errors.Errorf("stack '%s' already exists, use 'happy update %s' to update it", stackName, stackName)
+		return errors.Wrap(err, "failed one of the happy client validations")
 	}
 
-	err = util.ValidateGitTree(happyConfig.GetProjectRoot())
+	// update the newly created stack
+	stack, err := happyClient.StackService.GetStack(ctx, stackName)
 	if err != nil {
-		logrus.Infof("failed to determine the state of the git tree: %s", err.Error())
+		return errors.Wrapf(err, "stack %s doesn't exist; this should never happen", stackName)
 	}
+	return updateStack(ctx, cmd, stack, force, happyClient)
+}
 
-	// if creating tag and none specified, generate the default tag
-	if createTag && (tag == "") {
-		tag, err = backend.GenerateTag(ctx)
-		if err != nil {
-			return err
+func validateECRExists(ctx context.Context, stackName string, dryRun bool, ecrTargetPathFormat string, happyClient *HappyClient) validation {
+	return func() error {
+		if !happyClient.HappyConfig.GetFeatures().EnableECRAutoCreation {
+			return nil
 		}
-	}
+		// TODO: this works, but need a game plan for backwards compability
+		// this has a strong coupling with the TF version that we are using,
+		// so if the user isn't on it yet, this will fail
+		allServicesHaveECR, err := (func() (bool, error) {
+			services, err := happyClient.ArtifactBuilder.GetECRsForServices(ctx)
+			if err != nil {
+				return false, err
+			}
+			for _, service := range happyClient.HappyConfig.GetServices() {
+				if _, ok := services[service]; !ok {
+					return false, nil
+				}
+			}
 
-	// if creating tag, build and push images
-	if createTag {
-		// parity with update command
-		buildOpts = append(buildOpts, artifact_builder.WithTags(tag))
-		err = ab.BuildAndPush(ctx, buildOpts...)
+			return true, nil
+		})()
 		if err != nil {
-			return err
+			return errors.Wrap(err, "unable to get ECRs for service")
```

Comment:
Might be useful to include the name of the service for which we cannot get the ECR